db/db_impl_secondary.cc

```diff
@@ -271,31 +270,28 @@ Status DBImplSecondary::RecoverLogFiles(
             iter->second = log_number;
           }
         }
+        auto last_sequence = *next_sequence - 1;
+        if ((*next_sequence != kMaxSequenceNumber) &&
+            (versions_->LastSequence() <= last_sequence)) {
+          versions_->SetLastAllocatedSequence(last_sequence);
+          versions_->SetLastPublishedSequence(last_sequence);
+          versions_->SetLastSequence(last_sequence);
+        }
       } else {
         // We are treating this as a failure while reading since we read valid
         // blocks that do not form coherent data
         reader->GetReporter()->Corruption(record.size(), status);
-        continue;
```

Comment:
if status is not OK, do we want to skip to next record? I understand there is no difference right now since it's the last line in the while loop but we may add more lines in the future

Reply:
Fair point, but I think we can add the continue when that happens. Right now, this `continue` is not doing anything.

Reply:
Plus, even if we keep the continue, we need to set the status back to OK. Otherwise, we will break out of the loop, too.

Reply:
sounds good